docs/TESTING.md

```diff
@@ -1,53 +1,60 @@
-this file will need minor edits for your btc configuration).
+```
+rpcuser=bitcoinrpc
+rpcpassword=123456abcdef
+fallbackfee=0.0002
+```
 
-### Running tests of sendpayment and tumbler (including with malicious makers)
+(any random password is fine of course). It is also advisable to wipe ~/.bitcoin/regtest first, in case it gets large and slow to process.
 
-(As well as the below, there is a rudimentary "single, automatic, end-to-end coinjoin test" in the file test/test_full_coinjoin.py. Recommended to run it
-with -s to see log output; what's "rudimentary" here is that errors may simply result in the process hanging, so you'll want to investigate in that case.
-Use command line:
+Then copy the `regtest_joinmarket.cfg` file from the `test/` directory to the root directory, you probably won't need to change anything here except perhaps the above password, and the `native` setting if you're doing bech32 wallet tests.
 
-    pytest --btcconf=/path/to/bitcoin.conf --btcroot=/path/to/bitcoin/bin/ --btcpwd=whatever --nirc=2 test/test_full_coinjoin.py -s
+Run the test suite via pytest:
 
-)
+    (jmvenv)$ pytest --btcconf=/path/to/bitcoin.conf --btcroot=/path/to/bitcoin/bin/ --btcpwd=whatever --nirc=2 --ignore test/test_full_coinjoin.py -p no:warnings
+
+#### Running tests of sendpayment and tumbler (including with malicious makers)
 
 The file `test/ygrunner.py` provides the ability to spin up a set of yieldgenerator
 bots against the local IRC instance with the local regtest blockchain. It can be
 started with
 
     pytest --btcroot=/path/to/bitcoin/bin/ --btcpwd=123456abcdef --nirc=2 test/ygrunner.py -s
 
-Here the `-s` flag is useful because it prints log output to the console. If you
+Here the `-s` flag is necessary because it prints log output to the console. If you
 keep the logging level at the default `INFO` only a minimum amount will come out, if
 you want more then enter this into the `joinmarket.cfg` in the root directory:
 
     [LOGGING]
     console_log_level = DEBUG
 
-It will print out a hex seed for a wallet you can use for tumble/sendpayment. Next,
-go into the `scripts/` directory and make sure you have copied the `regtest_joinmarket.cfg`
+It will print out a hex seed for a wallet you can use for tumble/sendpayment (just set the walletname to that hex value).
+Next, go into the `scripts/` directory and make sure you have copied the `regtest_joinmarket.cfg`
 file into that directory also, make any changes needed (like the LOGGING one above),
-and run either sendpayment or tumbler with whatever parameters you choose.
+and run either sendpayment or tumbler with whatever parameters you choose, BUT: remember to add the `--datadir=.` argument so that your test `joinmarket.cfg` file gets picked up, not the one in `~/.joinmarket`.
+
+So for example:
+
+```
+(jmvenv) $python sendpayment.py -N2 -m1 5c88acf2546bd7b083b9cfb2e0af7f2d --datadir=. 30000000 2Address
```

Comment:
Here and below maybe add space between `$` and `python`? Took me some time to understand it's just meant as a command prompt without space after $, not `$python` (as a shell variable).

Reply:
Done in https://github.com/JoinMarket-Org/joinmarket-clientserver/commit/29875b4a2ae151afbc3223271c8c7d5c325f0e46

Reply:
Sorry I forgot to press save, done in https://github.com/JoinMarket-Org/joinmarket-clientserver/pull/664/commits/b9ad12bc89b927160b51b9f7e7ff1e1807f93dca